release-docker.sh

```diff
@@ -1,14 +1,25 @@
-#!/bin/bash
+#!/bin/sh
 
-tag=$(git describe --exact-match 2>/dev/null)
-if [[ $? != 0 ]]; then
-	tag=latest
+set -eu
+
+if ! tag="$(git describe --exact-match 2> "/dev/null")"; then
```

Comment:
hm, I prefer the older form, as it's more obvious. I don't like expressions which are also statements.

In a way, it reminds me of Go's `if foo := bar; expr... {`, where the definition is only part of the nested scope. So I find it pretty confusing that it actually modifies the parent scope.

Reply:
I think that's purely personal, as what you are familiar with. But if you prefer the earlier, that's fine as well. I like this style better as it doesn't rely on having to manually deal with `$?`